Stem cells atac seq
================
Angela Simeone

Three distinct cell types have been profiled here: hESC, CNCC and NSC. Often we refer to hESC as simply ESC. <br /> <br />

For each cell, 3 biological replicates have been performed for a total of 9 libraries.

<table style="width:100%;">
<colgroup>
<col width="3%" />
<col width="5%" />
<col width="10%" />
<col width="4%" />
<col width="20%" />
<col width="14%" />
<col width="14%" />
<col width="14%" />
<col width="14%" />
</colgroup>
<thead>
<tr class="header">
<th>sample</th>
<th>cell</th>
<th>cell-extended</th>
<th>derived_from</th>
<th>peaks</th>
<th>run1_read1</th>
<th>run1_read2</th>
<th>run2_read1</th>
<th>run2_read2</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Sample 29</td>
<td>CNCC_ATAC_100_bio1</td>
<td>cranial neural crest cells (CNCC)</td>
<td>H9</td>
<td>CNCC_ATAC_100_bio1.SLX-18056_SLX-18895.merged.sorted_peaks.narrowPeak</td>
<td>CNCC_ATAC_100_bio1_SLX-18352.i707_i508.r_1.fq.gz</td>
<td>CNCC_ATAC_100_bio1_SLX-18352.i707_i508.r_2.fq.gz</td>
<td>CNCC_ATAC_100_bio1_SLX-18379.i707_i508.r_1.fq.gz</td>
<td>CNCC_ATAC_100_bio1_SLX-18379.i707_i508.r_2.fq.gz</td>
</tr>
<tr class="even">
<td>Sample 30</td>
<td>CNCC_ATAC_100_bio2</td>
<td>cranial neural crest cells (CNCC)</td>
<td>H9</td>
<td>CNCC_ATAC_100_bio2.SLX-18056_SLX-18895.merged.sorted_peaks.narrowPeak</td>
<td>CNCC_ATAC_100_bio2_SLX-18352.i706_i517.r_1.fq.gz</td>
<td>CNCC_ATAC_100_bio2_SLX-18352.i706_i517.r_2.fq.gz</td>
<td>CNCC_ATAC_100_bio2_SLX-18379.i706_i517.r_1.fq.gz</td>
<td>CNCC_ATAC_100_bio2_SLX-18379.i706_i517.r_2.fq.gz</td>
</tr>
<tr class="odd">
<td>Sample 31</td>
<td>CNCC_ATAC_100_bio3</td>
<td>cranial neural crest cells (CNCC)</td>
<td>H9</td>
<td>CNCC_ATAC_100_bio3.SLX-18056_SLX-18895.merged.sorted_peaks.narrowPeak</td>
<td>CNCC_ATAC_100_bio3_SLX-18352.i705_i502.r_1.fq.gz</td>
<td>CNCC_ATAC_100_bio3_SLX-18352.i705_i502.r_2.fq.gz</td>
<td>CNCC_ATAC_100_bio3_SLX-18379.i705_i502.r_1.fq.gz</td>
<td>CNCC_ATAC_100_bio3_SLX-18379.i705_i502.r_2.fq.gz</td>
</tr>
<tr class="even">
<td>Sample 32</td>
<td>ESC_ATAC_50_bio1</td>
<td>human embryonic stem cells (ESC)</td>
<td>H9</td>
<td>ESC_ATAC_50_bio1.SLX-18056_SLX-18895.merged.sorted_peaks.narrowPeak</td>
<td>ESC_ATAC_50_bio1_SLX-18352.i704_i505.r_1.fq.gz</td>
<td>ESC_ATAC_50_bio1_SLX-18352.i704_i505.r_2.fq.gz</td>
<td>ESC_ATAC_50_bio1_SLX-18379.i704_i505.r_1.fq.gz</td>
<td>ESC_ATAC_50_bio1_SLX-18379.i704_i505.r_2.fq.gz</td>
</tr>
<tr class="odd">
<td>Sample 33</td>
<td>ESC_ATAC_50_bio2</td>
<td>human embryonic stem cells (ESC)</td>
<td>H9</td>
<td>ESC_ATAC_50_bio2.SLX-18056_SLX-18895.merged.sorted_peaks.narrowPeak</td>
<td>ESC_ATAC_50_bio2_SLX-18352.i703_i506.r_1.fq.gz</td>
<td>ESC_ATAC_50_bio2_SLX-18352.i703_i506.r_2.fq.gz</td>
<td>ESC_ATAC_50_bio2_SLX-18379.i703_i506.r_1.fq.gz</td>
<td>ESC_ATAC_50_bio2_SLX-18379.i703_i506.r_2.fq.gz</td>
</tr>
<tr class="even">
<td>Sample 34</td>
<td>ESC_ATAC_50_bio4</td>
<td>human embryonic stem cells (ESC)</td>
<td>H9</td>
<td>ESC_ATAC_50_bio4.SLX-18056_SLX-18895.merged.sorted_peaks.narrowPeak</td>
<td>ESC_ATAC_50_bio4_SLX-18352.i711_i503.r_1.fq.gz</td>
<td>ESC_ATAC_50_bio4_SLX-18352.i711_i503.r_2.fq.gz</td>
<td>ESC_ATAC_50_bio4_SLX-18379.i711_i503.r_1.fq.gz</td>
<td>ESC_ATAC_50_bio4_SLX-18379.i711_i503.r_2.fq.gz</td>
</tr>
<tr class="odd">
<td>Sample 35</td>
<td>NSC_ATAC_50_bio2</td>
<td>neural stem cells (NSC)</td>
<td>H9</td>
<td>NSC_ATAC_50_bio2.SLX-18056_SLX-18895.merged.sorted_peaks.narrowPeak</td>
<td>NSC_ATAC_50_bio2_SLX-18352.i705_i508.r_1.fq.gz</td>
<td>NSC_ATAC_50_bio2_SLX-18352.i705_i508.r_2.fq.gz</td>
<td>NSC_ATAC_50_bio2_SLX-18379.i705_i508.r_1.fq.gz</td>
<td>NSC_ATAC_50_bio2_SLX-18379.i705_i508.r_2.fq.gz</td>
</tr>
<tr class="even">
<td>Sample 36</td>
<td>NSC_ATAC_50_bio3</td>
<td>neural stem cells (NSC)</td>
<td>H9</td>
<td>NSC_ATAC_50_bio3.SLX-18056_SLX-18895.merged.sorted_peaks.narrowPeak</td>
<td>NSC_ATAC_50_bio3_SLX-18352.i704_i504.r_1.fq.gz</td>
<td>NSC_ATAC_50_bio3_SLX-18352.i704_i504.r_2.fq.gz</td>
<td>NSC_ATAC_50_bio3_SLX-18379.i704_i504.r_1.fq.gz</td>
<td>NSC_ATAC_50_bio3_SLX-18379.i704_i504.r_2.fq.gz</td>
</tr>
<tr class="odd">
<td>Sample 37</td>
<td>NSC_ATAC_50_bio4</td>
<td>neural stem cells (NSC)</td>
<td>H9</td>
<td>NSC_ATAC_50_bio4.SLX-18056_SLX-18895.merged.sorted_peaks.narrowPeak</td>
<td>NSC_ATAC_50_bio4_SLX-18352.i710_i507.r_1.fq.gz</td>
<td>NSC_ATAC_50_bio4_SLX-18352.i710_i507.r_2.fq.gz</td>
<td>NSC_ATAC_50_bio4_SLX-18379.i710_i507.r_1.fq.gz</td>
<td>NSC_ATAC_50_bio4_SLX-18379.i710_i507.r_2.fq.gz</td>
</tr>
</tbody>
</table>

General basic processing of the data includes:

-   quality controls
-   adaptor trimming
-   alignment = and then merge sequencing runs
-   stats generation and genome-wide track (bigWig)
-   peak calling
-   identification of consensus regions per cell type

Processing
----------

fastQC check

``` bash
# === fastqc quality check QC ===
runs=(SLX-18379 SLX-18352)
for run in ${runs[@]}
do
  cd /stem_cells/Atac_seq/${run}
  mkdir fastqc
  for file in *.fq.gz
  do
    sbatch -o %j.out -e %j.err --mem 4G --wrap "fastqc --noextract --nogroup -q -o fastqc/ $file"
  done
done
```

Cut TruSeq Illumina adapters trimming adapters (cutadapt for paired-end)

``` bash
## === cut illumina adapters ===
for fq1 in *r_1.fq.gz
do
  fq2=${fq1%%r_1.fq.gz}r_2.fq.gz
  ls $fq1
  ls $fq2
  sbatch -o %j.out -e %j.err --mem 16000 --wrap "cutadapt -a AGATCGGAAGAGC -A AGATCGGAAGAG -o ${fq1%%.fq.gz}.trimmed.fq.gz -p ${fq2%%.fq.gz}.trimmed.fq.gz $fq1 $fq2 "
done
```

Alignment to hg38 (paired-end)

``` bash
# === aligning ===
# do this in both folders
cd /stem_cells/Atac_seq/SLX-18379

cd /stem_cells/Atac_seq/SLX-18379
g='/reference_genomes/hg38/hg38_selected.fa'
w='/reference_genomes/hg38/hg38.whitelist.sorted.bed'

mkdir aligned
for fq1_trimmed in *r_1.trimmed.fq.gz
do
  fq2_trimmed=${fq1_trimmed%%r_1.trimmed.fq.gz}r_2.trimmed.fq.gz
  echo sbatch -o %j.out -e %j.err --mem 16G --wrap="bwa mem -M -t 12 $g $fq1_trimmed $fq2_trimmed | samtools view -Sb -F780 -q 10 -L $w - > aligned/${fq1_trimmed%%_R1_001.trimmed.fq.gz}.hg38.bam"
  sbatch -o %j.out -e %j.err --mem 16G --wrap="bwa mem -M -t 12 $g $fq1_trimmed $fq2_trimmed | samtools view -Sb -F780 -q 10 -L $w - > aligned/${fq1_trimmed%%.r_1.trimmed.fq.gz}.hg38.bam"
done

# === sort bam ====
cd /stem_cells/Atac_seq/SLX-18379/aligned
for file in *bam
  do
  sbatch --mem 16G --wrap "samtools sort -@ 8 $file -o ${file/.bam/.sorted.bam}"
done
```

Remove duplicates

``` bash

 # === remove duplicates ====
cd /stem_cells/Atac_seq/SLX-18379/aligned
for file in *.sorted.bam 
do
  sbatch --mem 16G  --wrap "java -Xmx7g -jar /home/simeon01/applications/picard-2.20.3.jar MarkDuplicatesWithMateCigar INPUT=$file OUTPUT=${file%%.bam}.markduplicates.bam REMOVE_DUPLICATES=true  AS=true METRICS_FILE=${file%%.bam}.markduplicates_metrics.txt"
  echo "==========="
  echo "          "
done
```

Generate stats about total number of reads and total number of unique reads

``` bash
cd /stem_cells/Atac_seq/SLX-18379/aligned
for file in *.sorted.bam
  do
  bam_hg38=$file
  #bam_f_nodup=${bam%%.hg38.sort.bam}.hg38.sort.markduplicates.bam
  bam_hg38_nodup=${file%%.bam}.markduplicates.bam
  
  ls -lth $bam_hg38
  echo "====="
  ls -lth $bam_hg38_nodup
  
  sbatch  --mem 4G --wrap "samtools view -c -F 260 $bam_hg38 >  ${file%%.bam}.stat2"
  sbatch  --mem 4G --wrap "samtools view -c -F 260 $bam_hg38_nodup >  ${file%%.bam}.stat5"
done


touch atac_seq_katie_SLX-18379.out.stat
for bam in *.sorted.bam
  do
  base_name=${bam/.bam/}
  files_stat=`ls *.stat* | grep $base_name`
  echo $bam > temp
  cat $files_stat >> temp
  paste -d '\t' atac_seq_katie_SLX-18379.out.stat temp >> temp2
  mv temp2 atac_seq_katie_SLX-18379.out.stat
done
```

Generate tracks (bw)

``` bash
cd /stem_cells/Atac_seq/SLX-18379/aligned
for bam in *.markduplicates.bam
    do
  sbatch -o %j.out.log --mem 16G --wrap "samtools index $bam"
done

for bam in *.markduplicates.bam
    do
        tot_r_hg38=`cat ${bam%%.markduplicates.bam}.stat5`
        scal_factor_hg38=$(awk -v m=$tot_r_hg38 'BEGIN { print 1000000/m }')
        echo $scal_factor_hg38
      echo sbatch --mem 16G --wrap "bamCoverage --scaleFactor $scal_factor_hg38 -bs 10 -b $bam -of \"bigwig\" -o ${bam%%.markduplicates.bam}.w10.rpm.bw"
        sbatch --mem 16G --wrap "bamCoverage --scaleFactor $scal_factor_hg38 -bs 10 -b $bam -of \"bigwig\" -o ${bam%%.markduplicates.bam}.w10.rpm.bw"
        echo "=============="
done
```

Call peaks (macs2)

``` bash
cd /stem_cells/Atac_seq/SLX-18379/aligned
# === call peaks on atac-seq libraries where Chromosome M is excluded
mkdir macs2_output
biorep=(1 2 3)
cells=(CNCC_ATAC ESC_ATAC NSC_ATAC)

for cell in ${cells[@]}
do
  list_rep=`ls $cell**markduplicates.bam`
  for rep in ${list_rep[@]}
  do
    t_bam=$rep
    t_tmpBam=${t_bam%%.markduplicates.bam}.markduplicates.temp
    ls $t_bam
    sbatch --mem 16G --wrap="samtools view -h -F1024 $t_bam | grep -v -P '\tchrM\t' | samtools view -b - > $t_tmpBam && macs2 callpeak --keep-dup all -t $t_tmpBam -n macs2_output/${t_bam%%.markduplicates.bam}"
    echo "============"
  done
done
```

Fragment size distribution

``` bash
cd /stem_cells/Atac_seq/SLX-18379/aligned
for file in *.markduplicates.bam
do
  #/Users/marsic01/applications/picard-2.8.2.jar MarkDuplicates ==> this was the one used before
  # now picard in home
  #picard MarkDuplicates --version
  #2.18.12-SNAPSHOT
  sbatch --mem 16G  --wrap "java -Xmx7g -jar /home/simeon01/applications/picard-2.20.3.jar CollectInsertSizeMetrics INPUT=$file OUTPUT=${file%%.markduplicates.bam}.insert_size_metrics.txt H=${file%%.markduplicates.bam}.insert_size_metrics.pdf HISTOGRAM_WIDTH=1000"
  echo "==========="
  echo "          "
done
```

Check percentage reads mapping to chrM

``` bash
## count reads mapping to chrM
for file in *.markduplicates.bam
do
  sbatch --mem 12G --wrap "samtools idxstats $file > ${file%%.bam}.chr_distr.txt"
done
```

Merge bams from the 2 sequencing runs performed using the same libraries (to increase depth).

``` bash

path_run1=/stem_cells/Atac_seq/SLX-18352/aligned
path_run2=/stem_cells/Atac_seq/SLX-18379/aligned

merged_runs=/stem_cells/Atac_seq/merged_SLX-18379_SLX-18352
mkdir $merged_runs

cd $path_run2

list_experiments=`ls *hg38.sorted.bam`

for exp in ${list_experiments[@]}
do
  base_name=${exp%%_SLX*}
  #echo $base_name
  bam1=`ls $path_run1/$base_name*.hg38.sorted.bam`
  bam2=`ls $path_run2/$base_name*.hg38.sorted.bam`
  ls -lth $bam1
  ls -lth $bam2
  out_bam=$merged_runs/$base_name.SLX-18056_SLX-18895.merged.bam
  cmd="samtools merge -@ 4 $out_bam $bam1 $bam2"
  sbatch --mem 16G --wrap "$cmd"
  echo $out_bam
  echo $cmd
  echo " ============== "
done
  
for file in *bam
  do
  sbatch --mem 16G --wrap "samtools sort -@ 8 $file -o ${file/.bam/.sorted.bam}"
  sbatch --mem 16G --wrap "samtools sort -@ 8 -n $file -o ${file/.bam/.sortedByName.bam}"
done
```

Remove duplicates

``` bash
# === remove duplicates ====
cd /stem_cells/Atac_seq/merged_SLX-18379_SLX-18352
for file in *.merged.sorted.bam 
do
  sbatch --mem 16G  --wrap "java -Xmx7g -jar /home/simeon01/applications/picard-2.20.3.jar MarkDuplicatesWithMateCigar INPUT=$file OUTPUT=${file%%.bam}.markduplicates.bam REMOVE_DUPLICATES=true  AS=true METRICS_FILE=${file%%.bam}.markduplicates_metrics.txt"
  echo "==========="
  echo "          "
done
```

Call peaks (macs2)

``` bash

## call peaks with genriche
mkdir genrich_out
cells=(CNCC_ATAC ESC_ATAC NSC_ATAC)
for cell in ${cells[@]}
do
  list_rep=`ls $cell*.sortedByName.bam`
  for rep in ${list_rep[@]}
  do
    t_bam=$rep
    sbatch --mem 32G --wrap="Genrich  -t $t_bam -o genrich_out/${t_bam%%.bam}.narrowPeak  -j  -y  -r  -e chrM  -v"
    done
done

cd /stem_cells/Atac_seq/merged_SLX-18379_SLX-18352
mkdir macs2_output
mkdir macs2_output_q1
cells=(CNCC_ATAC ESC_ATAC NSC_ATAC)

for cell in ${cells[@]}
do
  list_rep=`ls $cell**markduplicates.bam`
  for rep in ${list_rep[@]}
  do
    t_bam=$rep
    t_tmpBam=${t_bam%%.markduplicates.bam}.markduplicates.temp
    ls $t_bam
    #sbatch --mem 16G --wrap="samtools view -h -F1024 $t_bam | grep -v -P '\tchrM\t' | samtools view -b - > $t_tmpBam && macs2 callpeak --keep-dup all -t $t_tmpBam -n macs2_output/${t_bam%%.markduplicates.bam}"
    sbatch --mem 16G --wrap="samtools view -h -F1024 $t_bam | grep -v -P '\tchrM\t' | samtools view -b - > $t_tmpBam && macs2 callpeak --keep-dup all -t $t_tmpBam --qvalue 1 -n macs2_output_q1/${t_bam%%.markduplicates.bam}.q1"
    echo "============"
  done
done
```

Generate consensus per cell

``` bash
cells=(CNCC_ATAC ESC_ATAC NSC_ATAC)

for cell in ${cells[@]}

do
  peak_files=`ls $cell*narrowPeak`
  multiIntersectBed -i $peak_files | awk '{if ($4>=2) print $0}' | sortBed -i - |  mergeBed -i - > $cell.multi2.bed
done
```

Generate stats

``` bash

for file in *.sorted.bam
  do
  bam_hg38=$file
  bam_hg38_nodup=${file%%.bam}.markduplicates.bam
  
  ls -lth $bam_hg38
  echo "====="
  ls -lth $bam_hg38_nodup
  
  sbatch  --mem 4G --wrap "samtools view -c -F 260 $bam_hg38 >  ${file%%.bam}.stat2"
  sbatch  --mem 4G --wrap "samtools view -c -F 260 $bam_hg38_nodup >  ${file%%.bam}.stat5"
done

for bam in *.markduplicates.bam
    do
  sbatch -o %j.out.log --mem 16G --wrap "samtools index $bam"
done
```

Generate tracks (BW)

``` bash

for bam in *.markduplicates.bam
    do
        tot_r_hg38=`cat ${bam%%.markduplicates.bam}.stat5`
        scal_factor_hg38=$(awk -v m=$tot_r_hg38 'BEGIN { print 1000000/m }')
        echo $scal_factor_hg38
      echo sbatch --mem 16G --wrap "bamCoverage --scaleFactor $scal_factor_hg38 -bs 10 -b $bam -of \"bigwig\" -o ${bam%%.markduplicates.bam}.w10.rpm.bw"
        sbatch --mem 16G --wrap "bamCoverage --scaleFactor $scal_factor_hg38 -bs 10 -b $bam -of \"bigwig\" -o ${bam%%.markduplicates.bam}.w10.rpm.bw"
        echo "=============="
done
```

Fragment size distribution

``` bash

for file in *.markduplicates.bam
do
  #/Users/marsic01/applications/picard-2.8.2.jar MarkDuplicates ==> this was the one used before
  # now picard in home
  #picard MarkDuplicates --version
  #2.18.12-SNAPSHOT
  sbatch --mem 16G  --wrap "java -Xmx7g -jar /home/simeon01/applications/picard-2.20.3.jar CollectInsertSizeMetrics INPUT=$file OUTPUT=${file%%.markduplicates.bam}.insert_size_metrics.txt H=${file%%.markduplicates.bam}.insert_size_metrics.pdf HISTOGRAM_WIDTH=1000"
  echo "==========="
  echo "          "
done
```

Fenerate tracks after merging runs

``` bash

for bam in *.markduplicates.bam
    do
  sbatch -o %j.out.log --mem 16G --wrap "samtools index $bam"
done

for bam in *.markduplicates.bam
    do
        tot_r_hg38=`cat ${bam%%.markduplicates.bam}.stat5`
        scal_factor_hg38=$(awk -v m=$tot_r_hg38 'BEGIN { print 1000000/m }')
        echo $scal_factor_hg38
      echo sbatch --mem 16G --wrap "bamCoverage --scaleFactor $scal_factor_hg38 -bs 10 -b $bam -of \"bigwig\" -o ${bam%%.markduplicates.bam}.w10.rpm.bw"
        sbatch --mem 16G --wrap "bamCoverage --scaleFactor $scal_factor_hg38 -bs 10 -b $bam -of \"bigwig\" -o ${bam%%.markduplicates.bam}.w10.rpm.bw"
        echo "=============="
done
```

Density plots at G4 specific and common sites
---------------------------------------------

We want to see what is the atac signal at specific and common G4 regions (when occurring at promoters = 1kb aroun TSS)

I started with general regions: - specific to ESC (compared to the other 2 cell types) - specific to NSC - specific to CNCC - common to all 3 cell types

``` bash

cd /stem_cells/Atac_seq/merged_SLX-18379_SLX-18352
folder_bed=/scratcha/sblab/simeon01/Data/20200326_extern_stem_cells_data/final_reference_sets_stem_cells_published_data
folder_out_result=/stem_cells/Atac_seq/merged_SLX-18379_SLX-18352/density_plots_atac_seq_at_G4_ppromoters
mkdir $folder_out_result
cd /stem_cells/Atac_seq/merged_SLX-18379_SLX-18352
for file in *bw
do
  sbatch --mem 12G --wrap="computeMatrix reference-point -S ${file} -R $folder_bed/*prom.sorted.bed  -a 1000 -b 1000 --outFileName $folder_out_result/${file}.matrix.gz --missingDataAsZero && 
  plotProfile --matrixFile $folder_out_result/${file}.matrix.gz -o $folder_out_result/${file}.matrix.pdf --averageType median --plotWidth 18 --plotHeight 10 &&
  plotHeatmap --matrixFile $folder_out_result/${file}.matrix.gz -o $folder_out_result/${file}.matrix_heatmap.pdf"
  echo "========================"
done

/scratcha/sblab/simeon01/Data/20200326_extern_stem_cells_data/final_reference_sets_stem_cells_published_data/*prom.sorted.bed
```

In R perform the difference between the 2 matrices.

``` r
# subtract the two files
setwd('/Users/simeon01/Documents/Katie/Sequencing_Sept19_bio_multi2_compendium/broad/scenario1_multi2/density_plots_atac_seq_at_G4_ppromoters')
#NSC_ATAC*matrix*gz
#ESC_ATAC*matrix*gz
#CNCC_ATAC*matrix*gz

case_label_ESC='ESC_ATAC'

case_label_CNCC='CNCC_ATAC'

case_label_NSC='NSC_ATAC'


average_densities <- function(case_label){
  matrix_files <- list.files(".",paste0(case_label,".*.matrix.gz$"))
  matrix_files
  # extract params for initialize matrix
  header_temp <- readLines(gzfile(matrix_files[1]),n=1)
  
  A <- unlist(strsplit(header_temp, split=',\"'))
  A <- gsub('\"','',A)
  A <- gsub('[','',A, fixed = T)
  A <- gsub(']','',A, fixed = T)
  N_rows <- as.numeric(tail(unlist(strsplit(grep('group_boundaries',A, value =T),',|:')),1))
  groups_ind <- as.numeric(tail(unlist(strsplit(grep('group_boundaries',A, value =T),',|:')),-2)) # to exclude first 2
  N_col <- as.numeric(tail(unlist(strsplit(grep('sample_boundaries',A, value =T),',|:')),1))

  temp1_sum <- matrix(0,ncol=N_col,nrow=N_rows)
  for (i in 1:length(matrix_files)){
    header <- readLines(gzfile(matrix_files[i]),n=1)
    temp1 <- read.table(gzfile(matrix_files[i]),skip = 1, stringsAsFactors = F) 
    temp1_sum <- temp1_sum +  temp1[,7:dim(temp1)[2]]
    print(i)
  }
  temp1_avg <- temp1_sum/length(matrix_files)
  temp1_avg_complete <- cbind(temp1[,1:6],temp1_avg)
  file_name <- paste0(case_label,'_average.matrix')
  fileConn<-file(file_name)
  writeLines(header,fileConn)
  write.table(temp1_avg_complete, file = file_name, append = T,sep = "\t",quote = F, col.names = F, row.names = F)
  flag='done'
  close(fileConn)

  return(flag)
}  


average_densities(case_label_ESC)
average_densities(case_label_CNCC)
average_densities(case_label_NSC) 
```

Copy back to cluster

``` bash
scp /Users/simeon01/Documents/Katie/Sequencing_Sept19_bio_multi2_compendium/broad/scenario1_multi2/density_plots_atac_seq_at_G4_ppromoters/*average*matrix.gz simeon01@10.20.236.34:/stem_cells/Atac_seq/merged_SLX-18379_SLX-18352/density_plots_atac_seq_at_G4_ppromoters/
```

go to cluster and make heatmap
==============================

``` bash
cd /stem_cells/Atac_seq/merged_SLX-18379_SLX-18352/density_plots_atac_seq_at_G4_ppromoters
for file in *average*.matrix.gz
do
  sbatch --mem 12G --wrap="plotHeatmap --matrixFile $file  --averageTypeSummaryPlot median -o ${file%%.matrix.gz}.matrix_heatmap.pdf"
  echo "========================"
done

#locally 
scp simeon01@10.20.236.34:/stem_cells/Atac_seq/merged_SLX-18379_SLX-18352/density_plots_atac_seq_at_G4_ppromoters/*average.matrix_heatmap.pdf /Users/simeon01/Documents/Katie/Sequencing_Sept19_bio_multi2_compendium/broad/scenario1_multi2/density_plots_atac_seq_at_G4_ppromoters/.
```

Evaluate again the density but in this case only for pairwise comparisons

``` bash

# for regions that are common and specific, assign a sign (based on OQs) and use the coordinates of the center
cd /scratcha/sblab/simeon01/Data/Katie_high_level_analysis_Oct2019/scenario1_multi2/bed_intersections_common
path_oqs=/reference_genomes/OQs
for file in *C.bed
do
  sbatch --mem 2G --wrap "annotateBed -i $file -files $path_oqs/OQ_plus_hits.lifted_hg19_to_hg38.Minus_Strand_bed6.bed $path_oqs/OQ_minus_hits.lifted_hg19_to_hg38.Plus_Strand_bed6.bed -names Minus_Strand_bed6 Plus_Strand_bed6  -counts > ${file%%.bed}.G4_peaks_anotated_by_OQsStrand.bed"
done



cells=(CNCC NSC)
folder_out_result=/stem_cells/Atac_seq/merged_SLX-18379_SLX-18352/density_plots_second_atac_seq_at_G4_ppromoters
mkdir $folder_out_result

for cell in ${cells[@]}
do
cd /stem_cells/Atac_seq/merged_SLX-18379_SLX-18352
  for bwfile in ${cell}*bw
  do
    # DAUTHER CELL 1
    sbatch --mem 6G --wrap="computeMatrix reference-point -S ${bwfile} -R ${cell}_bed_lists/*bed  -a 1000 -b 1000 --outFileName $folder_out_result/${bwfile}_${cell}.matrix.gz --missingDataAsZero && 
    plotProfile --matrixFile $folder_out_result/${bwfile}_${cell}.matrix.gz -o $folder_out_result/${bwfile}_${cell}.matrix.pdf --averageType median --plotWidth 18 --plotHeight 10 &&
    plotHeatmap --matrixFile $folder_out_result/${bwfile}_${cell}.matrix.gz -o $folder_out_result/${bwfile}_${cell}.matrix_heatmap.pdf"
    echo "========================"
  done
  for bwfile in ESC*bw
  do
    # ESC
    sbatch --mem 6G --wrap="computeMatrix reference-point -S ${bwfile} -R ${cell}_bed_lists/*bed  -a 1000 -b 1000 --outFileName $folder_out_result/${bwfile}_${cell}.matrix.gz --missingDataAsZero && 
    plotProfile --matrixFile $folder_out_result/${bwfile}_${cell}.matrix.gz -o $folder_out_result/${bwfile}_${cell}.matrix.pdf --averageType median --plotWidth 18 --plotHeight 10 &&
    plotHeatmap --matrixFile $folder_out_result/${bwfile}_${cell}.matrix.gz -o $folder_out_result/${bwfile}_${cell}.matrix_heatmap.pdf"
    echo "========================"
  done
  echo "**********"
done
```

Locally merge bio rep

``` r
setwd('/Users/simeon01/Documents/Katie/Sequencing_Sept19_bio_multi2_compendium/broad/scenario1_multi2/density_plots_second_atac_seq_at_G4_ppromoters')
#NSC_ATAC*matrix*gz
#ESC_ATAC*matrix*gz
#CNCC_ATAC*matrix*gz

ending_label_CNCC="_CNCC.matrix.gz"
ending_label_NSC="_NSC.matrix.gz"

case_label_ESC='ESC_ATAC_50_bio'

case_label_CNCC='CNCC_ATAC_100_bio'

case_label_NSC='NSC_ATAC_50_bio'


average_densities <- function(case_label,ending_label){
  matrix_files <- list.files(".",paste0(case_label,".*",ending_label,"$"))
  matrix_files
  # extract params for initialize matrix
  header_temp <- readLines(gzfile(matrix_files[1]),n=1)
  
  A <- unlist(strsplit(header_temp, split=',\"'))
  A <- gsub('\"','',A)
  A <- gsub('[','',A, fixed = T)
  A <- gsub(']','',A, fixed = T)
  N_rows <- as.numeric(tail(unlist(strsplit(grep('group_boundaries',A, value =T),',|:')),1))
  groups_ind <- as.numeric(tail(unlist(strsplit(grep('group_boundaries',A, value =T),',|:')),-2)) # to exclude first 2
  N_col <- as.numeric(tail(unlist(strsplit(grep('sample_boundaries',A, value =T),',|:')),1))

  temp1_sum <- matrix(0,ncol=N_col,nrow=N_rows)
  for (i in 1:length(matrix_files)){
    header <- readLines(gzfile(matrix_files[i]),n=1)
    temp1 <- read.table(gzfile(matrix_files[i]),skip = 1, stringsAsFactors = F) 
    temp1_sum <- temp1_sum +  temp1[,7:dim(temp1)[2]]
    print(i)
  }
  temp1_avg <- temp1_sum/length(matrix_files)
  temp1_avg_complete <- cbind(temp1[,1:6],temp1_avg)
  file_name <- paste0(case_label,ending_label,'_average.matrix')
  fileConn<-file(file_name)
  writeLines(header,fileConn)
  write.table(temp1_avg_complete, file = file_name, append = T,sep = "\t",quote = F, col.names = F, row.names = F)
  flag='done'
  close(fileConn)

  return(flag)
}  


average_densities(case_label_ESC,ending_label_CNCC)
average_densities(case_label_ESC,ending_label_NSC)
average_densities(case_label_CNCC,ending_label_CNCC)
average_densities(case_label_NSC,ending_label_NSC) 
```

``` bash

scp *average.matrix*  simeon01@10.20.236.34:/stem_cells/Atac_seq/merged_SLX-18379_SLX-18352/density_plots_second_atac_seq_at_G4_ppromoters/

cd /stem_cells/Atac_seq/merged_SLX-18379_SLX-18352/density_plots_second_atac_seq_at_G4_ppromoters/

for file in *average*.matrix.gz
do
  sbatch --mem 12G --wrap="plotHeatmap --matrixFile $file  --averageTypeSummaryPlot median -o ${file%%.matrix.gz}.matrix_heatmap.pdf"
  echo "========================"
done

scp simeon01@10.20.236.34:/stem_cells/Atac_seq/merged_SLX-18379_SLX-18352/density_plots_second_atac_seq_at_G4_ppromoters/*average.matrix*pdf .
```

Check accessibility only for bivalent promoters and not

``` bash

folder_bivalent_prom=/scratcha/sblab/simeon01/Data/20200326_extern_stem_cells_data/final_reference_sets_stem_cells_published_data
# *bival_G4*sorted.bed # pattern for the bed


cells=(ESC CNCC NSC)
folder_out_result=/stem_cells/Atac_seq/merged_SLX-18379_SLX-18352/density_plots_second_atac_seq_at_bivalent_promoters
mkdir $folder_out_result

for cell in ${cells[@]}
do
cd /stem_cells/Atac_seq/merged_SLX-18379_SLX-18352
  for bwfile in *bw
  do
    # DAUTHER CELL 1
    sbatch --mem 6G --wrap="computeMatrix reference-point -S ${bwfile} -R $folder_bivalent_prom/${cell}*bival_G4*sorted.bed  -a 1000 -b 1000 --outFileName $folder_out_result/${bwfile}_${cell}.matrix.gz --missingDataAsZero && 
    plotProfile --matrixFile $folder_out_result/${bwfile}_${cell}.matrix.gz -o $folder_out_result/${bwfile}_${cell}.matrix.pdf --averageType median --plotWidth 18 --plotHeight 10 &&
    plotHeatmap --matrixFile $folder_out_result/${bwfile}_${cell}.matrix.gz -o $folder_out_result/${bwfile}_${cell}.matrix_heatmap.pdf"
    echo "========================"
  done
  echo "**********"
done


#copy folder locally to laptop to combine different replicates
#cd /Users/simeon01/Documents/Katie/Sequencing_Sept19_bio_multi2_compendium/broad/scenario1_multi2/
# scp -r simeon01@10.20.236.34:/stem_cells/Atac_seq/merged_SLX-18379_SLX-18352/density_plots_second_atac_seq_at_bivalent_promoters



folder_H3K4me3_prom=/scratcha/sblab/simeon01/Data/20200326_extern_stem_cells_data/final_reference_sets_stem_cells_published_data
# *bival_G4*sorted.bed # pattern for the bed


cells=(ESC CNCC NSC)
folder_out_result=/stem_cells/Atac_seq/merged_SLX-18379_SLX-18352/density_plots_second_atac_seq_at_H3K4me3_promoters
mkdir $folder_out_result

for cell in ${cells[@]}
do
cd /stem_cells/Atac_seq/merged_SLX-18379_SLX-18352
  for bwfile in *bw
  do
    # DAUTHER CELL 1
    sbatch --time 00:20:00 --mem 6G --wrap="computeMatrix reference-point -S ${bwfile} -R $folder_H3K4me3_prom/${cell}*_H3K4me3_G4*sorted.bed  -a 1000 -b 1000 --outFileName $folder_out_result/${bwfile}_${cell}.matrix.gz --missingDataAsZero && 
    plotProfile --matrixFile $folder_out_result/${bwfile}_${cell}.matrix.gz -o $folder_out_result/${bwfile}_${cell}.matrix.pdf --averageType median --plotWidth 18 --plotHeight 10 &&
    plotHeatmap --matrixFile $folder_out_result/${bwfile}_${cell}.matrix.gz -o $folder_out_result/${bwfile}_${cell}.matrix_heatmap.pdf"
    echo "========================"
  done
  echo "**********"
done

#copy folder locally to laptop to combine different replicates
#cd /Users/simeon01/Documents/Katie/Sequencing_Sept19_bio_multi2_compendium/broad/scenario1_multi2/
# scp -r simeon01@10.20.236.34:/stem_cells/Atac_seq/merged_SLX-18379_SLX-18352/density_plots_second_atac_seq_at_H3K4me3_promoters
```

Locally combine replicates in R

``` r
setwd('/Users/simeon01/Documents/Katie/Sequencing_Sept19_bio_multi2_compendium/broad/scenario1_multi2/density_plots_second_atac_seq_at_bivalent_promoters')
#NSC_ATAC*matrix*gz
#ESC_ATAC*matrix*gz
#CNCC_ATAC*matrix*gz

ending_label_CNCC="_CNCC.matrix.gz"
ending_label_NSC="_NSC.matrix.gz"
ending_label_ESC="_ESC.matrix.gz"

case_label_ESC='ESC_ATAC_50_bio'

case_label_CNCC='CNCC_ATAC_100_bio'

case_label_NSC='NSC_ATAC_50_bio'


average_densities <- function(case_label,ending_label){
  matrix_files <- list.files(".",paste0(case_label,".*",ending_label,"$"))
  matrix_files
  # extract params for initialize matrix
  header_temp <- readLines(gzfile(matrix_files[1]),n=1)
  
  A <- unlist(strsplit(header_temp, split=',\"'))
  A <- gsub('\"','',A)
  A <- gsub('[','',A, fixed = T)
  A <- gsub(']','',A, fixed = T)
  N_rows <- as.numeric(tail(unlist(strsplit(grep('group_boundaries',A, value =T),',|:')),1))
  groups_ind <- as.numeric(tail(unlist(strsplit(grep('group_boundaries',A, value =T),',|:')),-2)) # to exclude first 2
  N_col <- as.numeric(tail(unlist(strsplit(grep('sample_boundaries',A, value =T),',|:')),1))

  temp1_sum <- matrix(0,ncol=N_col,nrow=N_rows)
  for (i in 1:length(matrix_files)){
    header <- readLines(gzfile(matrix_files[i]),n=1)
    temp1 <- read.table(gzfile(matrix_files[i]),skip = 1, stringsAsFactors = F) 
    temp1_sum <- temp1_sum +  temp1[,7:dim(temp1)[2]]
    print(i)
  }
  temp1_avg <- temp1_sum/length(matrix_files)
  temp1_avg_complete <- cbind(temp1[,1:6],temp1_avg)
  file_name <- paste0(case_label,ending_label,'_average.matrix')
  fileConn<-file(file_name)
  writeLines(header,fileConn)
  write.table(temp1_avg_complete, file = file_name, append = T,sep = "\t",quote = F, col.names = F, row.names = F)
  flag='done'
  close(fileConn)
  list_sizes <- list("N_rows"=N_rows,"groups_ind"=groups_ind,"N_col"=N_col)
  return(list_sizes)
}  


average_densities(case_label_ESC,ending_label_ESC)
average_densities(case_label_ESC,ending_label_CNCC)
average_densities(case_label_ESC,ending_label_NSC)

average_densities(case_label_CNCC,ending_label_ESC)
average_densities(case_label_CNCC,ending_label_CNCC)
average_densities(case_label_CNCC,ending_label_NSC)

average_densities(case_label_NSC,ending_label_ESC) 
average_densities(case_label_NSC,ending_label_CNCC) 
average_densities(case_label_NSC,ending_label_NSC) 

## do the same for H3K4me3
setwd('/Users/simeon01/Documents/Katie/Sequencing_Sept19_bio_multi2_compendium/broad/scenario1_multi2/density_plots_second_atac_seq_at_H3K4me3_promoters')
#NSC_ATAC*matrix*gz
#ESC_ATAC*matrix*gz
#CNCC_ATAC*matrix*gz

ending_label_CNCC="_CNCC.matrix.gz"
ending_label_NSC="_NSC.matrix.gz"
ending_label_ESC="_ESC.matrix.gz"

case_label_ESC='ESC_ATAC_50_bio'

case_label_CNCC='CNCC_ATAC_100_bio'

case_label_NSC='NSC_ATAC_50_bio'


average_densities <- function(case_label,ending_label){
  matrix_files <- list.files(".",paste0(case_label,".*",ending_label,"$"))
  matrix_files
  # extract params for initialize matrix
  header_temp <- readLines(gzfile(matrix_files[1]),n=1)
  
  A <- unlist(strsplit(header_temp, split=',\"'))
  A <- gsub('\"','',A)
  A <- gsub('[','',A, fixed = T)
  A <- gsub(']','',A, fixed = T)
  N_rows <- as.numeric(tail(unlist(strsplit(grep('group_boundaries',A, value =T),',|:')),1))
  groups_ind <- as.numeric(tail(unlist(strsplit(grep('group_boundaries',A, value =T),',|:')),-2)) # to exclude first 2
  N_col <- as.numeric(tail(unlist(strsplit(grep('sample_boundaries',A, value =T),',|:')),1))

  temp1_sum <- matrix(0,ncol=N_col,nrow=N_rows)
  for (i in 1:length(matrix_files)){
    header <- readLines(gzfile(matrix_files[i]),n=1)
    temp1 <- read.table(gzfile(matrix_files[i]),skip = 1, stringsAsFactors = F) 
    temp1_sum <- temp1_sum +  temp1[,7:dim(temp1)[2]]
    print(i)
  }
  temp1_avg <- temp1_sum/length(matrix_files)
  temp1_avg_complete <- cbind(temp1[,1:6],temp1_avg)
  file_name <- paste0(case_label,ending_label,'_average.matrix')
  fileConn<-file(file_name)
  writeLines(header,fileConn)
  write.table(temp1_avg_complete, file = file_name, append = T,sep = "\t",quote = F, col.names = F, row.names = F)
  flag='done'
  close(fileConn)
  list_sizes <- list("N_rows"=N_rows,"groups_ind"=groups_ind,"N_col"=N_col)
  return(list_sizes)
}  


average_densities(case_label_ESC,ending_label_ESC)
average_densities(case_label_ESC,ending_label_CNCC)
average_densities(case_label_ESC,ending_label_NSC)

average_densities(case_label_CNCC,ending_label_ESC)
average_densities(case_label_CNCC,ending_label_CNCC)
average_densities(case_label_CNCC,ending_label_NSC)

average_densities(case_label_NSC,ending_label_ESC) 
average_densities(case_label_NSC,ending_label_CNCC) 
average_densities(case_label_NSC,ending_label_NSC) 





# cd /Users/simeon01/Documents/Katie/Sequencing_Sept19_bio_multi2_compendium/broad/scenario1_multi2/density_plots_second_atac_seq_at_bivalent_promoters

scp *average.matrix.gz simeon01@10.20.236.34:/stem_cells/Atac_seq/merged_SLX-18379_SLX-18352/density_plots_second_atac_seq_at_bivalent_promoters


cd /Users/simeon01/Documents/Katie/Sequencing_Sept19_bio_multi2_compendium/broad/scenario1_multi2/density_plots_second_atac_seq_at_H3K4me3_promoters

scp *average.matrix.gz simeon01@10.20.236.34:/stem_cells/Atac_seq/merged_SLX-18379_SLX-18352/density_plots_second_atac_seq_at_H3K4me3_promoters
```

on the Cluster

``` bash
cd /stem_cells/Atac_seq/merged_SLX-18379_SLX-18352/density_plots_second_atac_seq_at_bivalent_promoters
cd /stem_cells/Atac_seq/merged_SLX-18379_SLX-18352/density_plots_second_atac_seq_at_H3K4me3_promoters
for file in *average*.matrix.gz
do
  sbatch --mem 2G --time 00:10:00 --wrap="plotHeatmap --matrixFile $file  --averageTypeSummaryPlot median -o ${file%%.matrix.gz}.matrix_heatmap.pdf --yMax 1 --zMax 1"
  echo "========================"
done

#cd /Users/simeon01/Documents/Katie/Sequencing_Sept19_bio_multi2_compendium/broad/scenario1_multi2/density_plots_second_atac_seq_at_bivalent_promoters
#scp simeon01@10.20.236.34:/stem_cells/Atac_seq/merged_SLX-18379_SLX-18352/density_plots_second_atac_seq_at_bivalent_promoters/*gz_average.matrix_heatmap.pdf . 
```
